cybereason/icon_cybereason/util/api.py

```diff
@@ -28,20 +28,75 @@ def connect(self):
                 data="There is a problem with service. Please check your credentials and permissions",
             )
 
-    def file_search(self, server_filter: dict, file_filter: dict):
-        api_response = self.session.request(
+    def isolate_machines(self, malop_id: str, pylum_ids: [str]) -> dict:
+        return self.send_request(
             "POST",
-            f"{self.base_url}/rest/sensors/action/fileSearch",
-            json={"filters": server_filter, "fileFilters": file_filter},
-            headers=self.headers,
+            "/rest/monitor/global/commands/isolate",
+            payload={"malopId": malop_id, "pylumIds": pylum_ids},
         )
 
-        if api_response.status_code >= 400:
-            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=api_response.text)
-        if 200 <= api_response.status_code < 300:
-            return insightconnect_plugin_runtime.helper.clean(json.loads(api_response.content))
+    def remediate(self, initiator_user_name: str, actions_by_machine: dict) -> dict:
+        return self.send_request(
+            "POST",
+            "/rest/remediate",
+            payload={
+                "initiatorUserName": initiator_user_name,
+                "actionsByMachine": actions_by_machine,
+            },
+        )
+
+    def file_search(self, server_filter: dict, file_filter: dict) -> dict:
+        if not file_filter:
+            raise PluginException(
+                cause="File filter shouldn't be empty.", assistance="Please check this input."
+            )
+
+        return self.send_request(
+            "POST",
+            "/rest/sensors/action/fileSearch",
+            payload={"filters": server_filter, "fileFilters": file_filter},
+        )
+
+    def send_request(
+        self, method: str, path: str, params: dict = None, payload: dict = None
+    ) -> dict:
+        path = path.lstrip("/")
```

Comment:
I wouldn't have done this like this. You're stripping the / here and re-adding it below in line 68. 

Do this instead: 
https://stackoverflow.com/questions/8223939/how-to-join-absolute-and-relative-urls

`urlparse.urljoin(url1, url2)`